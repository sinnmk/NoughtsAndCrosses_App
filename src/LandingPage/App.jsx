import {Component} from 'react';
import AppTemplate from './AppTemplate';
import './App.css';
import Axios from 'axios';

export default class App extends Component { 
  constructor(props){
    super(props);
    this.state = {};
  }

  render() {
    return AppTemplate.call(this);
  }

  componentDidMount(){
    Axios.get(`http:localhost:52718/home`)
    .then(response =>{
      this.setState({
        content: response.data.App
      }) 
    })
  }
}
