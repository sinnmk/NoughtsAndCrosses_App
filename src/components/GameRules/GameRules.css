.game-rules-header{
    border-block-end-style: dashed; 
}
.game-body-container{
    background-color: cadetblue;
}