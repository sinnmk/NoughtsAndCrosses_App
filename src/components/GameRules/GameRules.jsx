import {Component} from 'react';
import GameRulesTemplate from './GameRulesTemplate';

export default class GameRules extends Component{

    constructor(props){
        super(props);
        this.state = {}
    }

    render(){
        return GameRulesTemplate.call(this);
    }
    
    backToMenu(){

    }
}
