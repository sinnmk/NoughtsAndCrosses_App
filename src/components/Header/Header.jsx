import {Component} from 'react';
import 'reactstrap'
import HeaderTemplate from './HeaderTemplate'

export default class Header extends Component{
    constructor(props){
        super(props);
        this.state = {}
    }

    render(){
        return HeaderTemplate.call(this);
    }
}