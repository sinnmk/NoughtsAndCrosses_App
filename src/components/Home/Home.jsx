import { Component } from 'react';
import 'reactstrap'
import HomeTemplate from './HomeTemplate';

export default class Home extends Component {
    constructor(props){
        super(props);
        this.state = {}
    }

    render() {
        return HomeTemplate.call(this);
    }
}