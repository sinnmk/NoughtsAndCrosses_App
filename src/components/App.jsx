import React, {Component} from 'react';
import './App.css';
import axios from 'axios';
import {Button} from 'reactstrap'
import {BrowserRouter as Router, Link, Route} from 'react-router-dom';
import Header from './Header/Header'
import Home from './Home/Home'

export default class App extends Component { 
  constructor(props){
    super();
    this.state = {}
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(){
    axios.get('http:localhost:64353/home')
      .then(response => this.setState({}))
  }

  render() {
  return(
    <Router>
      <div className="App">
        <Header/>
        <ul>
            <Link to="/" onClick={Home}><Button size="lg">Home</Button></Link>
        </ul>
        <hr />
          <Route exact path='/' component={Home}/>
      </div>
    </Router>
  )
  }
}
