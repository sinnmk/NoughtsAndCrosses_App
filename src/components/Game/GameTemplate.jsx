import React from 'react';
import {Button} from 'reactstrap';
import Home from '../Home/Home';
import {Link} from 'react-router-dom'

const GameTemplate = () =>{
    return (
        <div>
            <Link to="/" onClick={Home}><Button className = 'back-button'>Back to Main Menu</Button></Link>
        </div>
    )
}
export default GameTemplate;