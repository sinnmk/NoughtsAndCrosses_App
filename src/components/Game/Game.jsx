import { Component } from 'react';
import GameTemplate from './GameTemplate'; 

export default class Game extends Component {
    constructor (props) {
        super(props);
        this.onRadioBtnClick = this.onRadioBtnClick.bind(this);
      }
    
      onRadioBtnClick(rSelected) {
        this.setState({ rSelected });
      }
    
      render() {
        return GameTemplate.call(this);
      }
    }