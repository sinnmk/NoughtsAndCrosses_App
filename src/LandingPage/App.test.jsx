import React from 'react';
import ReactDOM from 'react-dom';
import {BrowserRouter as Router, Route, Link} from 'react-router-dom';
import renderer from 'react-test-renderer';
import App from './App';
import Header from '../components/Header/Header';
import Home from '../components/Home/Home';
import {Button} from 'reactstrap';
import Game from '../components/Game/Game';
import GameRules from '../components/GameRules/GameRules';
import Signup from '../components/Signup/Signup';

describe('When app renders', () =>{

  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('Link to playGame matches snapshot game', () => {
    const component = renderer.create(
      <Router>
        <div className="App">
          <Header/>
          <ul>
              <Link to="/" onClick={Home}><Button size="lg">Home</Button></Link>
              <Link to="/playgame" onClick={Game}><Button size="lg">Play Game</Button></Link>
              <Link to="/gamerules" onClick={GameRules}><Button size="lg">About Tic-Tac-Toe</Button></Link>
              <Link to="/signup" onClick={Signup}><Button size="lg">Sign Up</Button></Link>
          </ul>
          <hr />
            <Route exact path='/' component={Home}/>
            <Route path='/playgame' component={Game}/>
            <Route path='/gamerules' component={GameRules}/>
            <Route path='/signup' component={Signup}/>
        </div>
      </Router>
    ).toJSON();
    expect(component).toMatchSnapshot();
  })

})
