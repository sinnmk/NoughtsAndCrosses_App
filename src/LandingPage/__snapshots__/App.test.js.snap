// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When app renders Link to playGame matches snapshot game 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      TIC TAC TOE
    </h1>
  </div>
  <ul>
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary btn-lg"
        onClick={[Function]}
        type={undefined}
      >
        Home
      </button>
    </a>
    <a
      href="/playgame"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary btn-lg"
        onClick={[Function]}
        type={undefined}
      >
        Play Game
      </button>
    </a>
    <a
      href="/gamerules"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary btn-lg"
        onClick={[Function]}
        type={undefined}
      >
        About Tic-Tac-Toe
      </button>
    </a>
    <a
      href="/signup"
      onClick={[Function]}
    >
      <button
        className="btn btn-secondary btn-lg"
        onClick={[Function]}
        type={undefined}
      >
        Sign Up
      </button>
    </a>
  </ul>
  <hr />
  <div>
    Home
  </div>
</div>
`;
